### PR TITLE
[MM-59334] Migrate tooltips of "components/sidebar/sidebar_header/sidebar_header.tsx" to WithTooltip

### PR DESCRIPTION
**Pull request by Gesare5**
#### Summary

Replaces OverlayTrigger and Tooltip, with WithTooltip in sidebar_header component

#### Ticket Link

Fixes  [#27558](https://github.com/mattermost/mattermost/issues/27558)
Jira  [https://mattermost.atlassian.net/browse/MM-59334](https://mattermost.atlassian.net/browse/MM-59334)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/sidebar/sidebar_header/sidebar_header.tsx [hunk 1]
```diff
@@ -124,30 +121,27 @@ const SidebarHeader = (props: Props) => {
...
-                            usageDeltaTeams={usageDeltas.teams.active}
-                        />
-                    </MenuWrapper>
-                </OverlayTrigger>
+                    </WithTooltip>
+                    <MainMenu
+                        id='sidebarDropdownMenu'
+                        usageDeltaTeams={usageDeltas.teams.active}
+                    />
+                </MenuWrapper>
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/sidebar/sidebar_header/sidebar_header.tsx:144]: WithTooltips wraps MenuWrapper as well, can you cross check previous code?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/sidebar/sidebar_header/sidebar_header.tsx (hunk 1 above)
Comment (Gesare5) [reply webapp/channels/src/components/sidebar/sidebar_header/sidebar_header.tsx:144]: Yes I noticed that, but the issue was even before the migrate the tooltip would not show when MenuWrapper was included. @M-ZubairAhmed , so I wasn't sure whether to wrap it or not.

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/sidebar/sidebar_header/sidebar_header.tsx:144]: Oh do you mean with MainMenu under the withTooltip the tooltip doesnt work?

Comment (Gesare5) [reply webapp/channels/src/components/sidebar/sidebar_header/sidebar_header.tsx:144]: Yeah @M-ZubairAhmed , so MenuWrapper requires exactly 2 children according to the error I was getting, that means I can't put MainMenu in the WithTooltip, it just returns a blank page. 
If I wrap the MenuWrapper in the WithTooltip (like in the initial implementation), the tootltip just doesn't show up. This was also the case when it was using the OverlayTrigger.

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/sidebar/sidebar_header/sidebar_header.tsx:144]: Ah i see also the functionality is not affected here so i think approach is fine with me

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**